### DISCUSSION TITLE
keybind for down-line-or-complete not working in zshrc

### DISCUSSION THREAD

**Discussion by wolfo** [marlonrichert/zsh-autocomplete#203] [Q&A] [answered]
# Environment
* Zsh version: **5.7.1; 5.8**
* `zsh-autocomplete` version: **9c933f6c76ceb5a82d8288a7a342dec0becba031**

Please report if the bug occurs without config files:
**doesn't apply**

If not, gradually add lines from your config file to the subshell until the bug appears.

Which combination of config causes the bug to appear?

**minimum config in ~/zshrc-test/.zshrc:**
```
source /home/wolfo/.antigen/bundles/marlonrichert/zsh-autocomplete/zsh-autocomplete.plugin.zsh
bindkey '^I' down-line-or-select
```
`ZDOTDIR=~/zshrc-test HOME=$PWD zsh`

# Steps to reproduce

1. **source zsh-autocomplete in your zshrc**
2. `bindkey '^I' down-line-or-select` **in zshrc**

## Expected behavior
**When tab is pressed, autocomplete if there is only one result. If there are more, tab scrolls through all the options instead.
(This is what happens when I enter `bindkey '^I' down-line-or-select` manually once the shell is started.)**

## Actual behavior
**Pressing tab completes the first result, regardless of how many results there are. The only way to select a result is by pressing down, as is the default.**

**I'd strongly prefer if tab didn't autocomplete when there are multiple results, but instead keep pressing tab (or typing) until I get the desired selection. And obviously I don't want to issue bindkey manually on every zsh instance :-)**

**Comment by marlonrichert** [comment, marked as answer]
> I'd strongly prefer if tab didn't autocomplete when there are multiple results, but instead keep pressing tab (or typing) until I get the desired selection.

That’s already possible, [as described in the Readme](https://github.com/marlonrichert/zsh-autocomplete/blob/master/README.md#change-tab-behavior). Just add _one_ (not both) of the following two lines to your `~/.zshrc` file (depending on which type of behavior your prefer; see the linked section in the Readme for details):
```zsh
# To use menu selection:
zstyle ':autocomplete:tab:*' widget-style menu-select

# To cycle completions without using menu selection:
zstyle ':autocomplete:tab:*' widget-style menu-complete
```

**Comment by wolfo** [comment]
Hey, exactly what I want :-) Thanks for the help.

However, I found that neither of these two zstyle options have an effect if I enter them in a shell (even with just the two-liner zshrc above), but **only** when I put them in zshrc.

This threw me off, because I actually tried both options before (by issuing manually).

**Comment by marlonrichert** [comment]
That’s a limitation of Zsh’s line editor. Changing that behavior at runtime would require swapping the widget.
